sccameraview/src/main/java/com/example/sccameraview/Camera2View.java

```diff
@@ -369,6 +379,70 @@ public int compare(Size lhs, Size rhs) {
         }
     }
 
+    @Override
+    public void takePicture() {
+        imageReader = ImageReader.newInstance(previewSize.getWidth(), previewSize.getHeight(), ImageFormat.JPEG, 1);
```

Comment:
The preview size matches the size of the displayed view. However in most cases, the user wants to create a much bigger size for the actual image. The preview size is under full HD, but the actual image should be 12 megapixels (just an example).